test/e2e/inference_with_adapters_test.go

```diff
@@ -104,6 +123,34 @@ func validateAdapterAdded(workspaceObj *kaitov1alpha1.Workspace, deploymentName
 	})
 }
 
+func validateNoInitContainers(workspaceObj *kaitov1alpha1.Workspace, expectedInitContainers []corev1.Container) {
```

Comment:
Init containers don't get deleted per my understanding - they get marked completed. Could rename to `validateInitContainersCompleted`?

Reply:
nvm resolved offline that new pod is created - as long as we check new deployment has the correct # of init containers, we can remove this